Update journal vclock only in box.cc (#10138)
tarantool/tarantool: changelogs/unreleased/gh-9916-registration-invalid-lsn.md

## bugfix/replication

> * Fixed a bug that an anon replica register or a replica name assignment could

Reviewer: ```suggestion
* Fixed a bug when an anonymous replica register or a replica name assignment could
```

Author: Thanks, fixed!
